packages/@aws-cdk/aws-ec2/README.md

```diff
@@ -638,9 +638,35 @@ Here is an example of applying some configuration to an instance:
 
 ```ts
 new ec2.Instance(this, 'Instance', {
-  init: ec2.CloudFormationInit.fromElements(
-    ec2.InitCommand.shellCommand('/bin/true'),
-  ),
+  // Showing the most complex setup, if you have simpler requirements
+  // you can use `CloudFormationInit.fromElements()`.
+  init: ec2.CloudFormationInit.fromConfigSets({
+    configSets: {
+      // Applies the configs below in this order
+      default: ['yumPreinstall', 'config'],
+    },
+    configs: {
+      yumPreinstall: new ec2.InitConfig([
+        // Install an Amazon Linux package using yum
+        ec2.InitPackage.yum('git'),
+      ]),
+      config: new ec2.InitConfig([
+        // Create a JSON file from tokens (can also create other files)
+        ec2.InitFile.fromObject('/etc/stack.json', {
```

Comment:
Reading this I'm regretting this not being `ec2.InitElement.fileFromObject(...)`.

Although even reading that I'm not sure it would be better. Augh!

Reply:
Nothing like 6 weeks without seeing a piece of code to bring a new perspective (although not necessarily enlightenment). :D